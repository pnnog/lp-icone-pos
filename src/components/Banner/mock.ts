export default {
  img: 'https://source.unsplash.com/user/willianjusten/1040x495',
  reminder: true,
  title: 'Especialização implantodontia',
  date: '1/1/1',
  teacher: 'Dr. Sandro Bittencourt | Dr. Márcio Marchionni'
};
