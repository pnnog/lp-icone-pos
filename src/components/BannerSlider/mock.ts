export default [
  {
    reminder: true,
    title: 'Especialização implantodontia',
    img: 'https://source.unsplash.com/user/willianjusten/1600x800',
    date: '25/03/2023',
    teacher: 'Dr. Sandro Bittencourt | Dr. Márcio Marchionni'
  },

  {
    title: 'Aperfeiçoamento em implantodontia',
    img: 'https://source.unsplash.com/user/willianjusten/1440x496',
    date: '29/03/2023',
    teacher: 'Dr. Sandro Bittencourt | Dr. Márcio Marchionni'
  }
];
