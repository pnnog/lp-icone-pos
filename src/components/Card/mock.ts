export const mockPacientCards = [
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x400',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tempor pradadsasoidsent exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },

  {
    img: 'https://source.unsplash.com/user/willianjusten/400x401',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tdadasdempor proideaasdnt exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x402',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tdadasdempor pdddroident exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x403',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tsadasdadadddadasdempor proident exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x404',
    title: 'Irure ad reprehenderit dolore voluptate Lorem adasdasdasdtdadasdempor proident exercitation qui.',
    modal:
      'Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.Est do et esse enim aliquip elit est officia. Laborum ex et aliquip magna eu reprehenderit commodo sunt enim non do aute elit incididunt. Esse et laboris exercitation sit consectetur adipisicing pariatur incididunt dolor excepteur duis incididunt in ex. Laborum cillum proident nulla quis deserunt. Et eiusmod laboris excepteur laboris labore dolore. Voluptate velit ullamco ea enim reprehenderit eiusmod occaecat nisi tempor proident amet eu fugiat sit. Aute occaecat non nisi officia voluptate.'
  }
];

export const mockCourseCards = [
  {
    img: 'https://source.unsplash.com/user/willianjusten/400x400',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tempor proident exercitation qsdasdasdui.'
  },

  {
    img: 'https://source.unsplash.com/user/willianjusten/400x401',
    title: 'Irure ad reprehenderit dolore voluptate Lorem tempor proident exercitadasdasdation qui.'
  }
];
