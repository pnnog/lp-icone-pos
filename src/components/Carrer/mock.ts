export default {
  img: '/img/gustavo-bastos.jpg',
  name: 'Gustavo Bastos',
  role: 'Cordenador',
  description:
    'Cillum voluptate consectetur nostrud proident deserunt culpa laboris incididunt quis sunt cillum. Dolor aliquip sunt pariatur ullamco labore labore velit sit dolore mollit elit proident. Amet consequat quis ipsum anim deserunt eiusmod. Tempor qui anim cillum voluptate nulla id minim sit. Consectetur irure ex cupidatat officia laboris dolor id pariatur fugiat ut voluptate non. Ea reprehenderit tempor officia eiusmod ex laborum amet incididunt ea culpa ullamco ad ipsum ut. Quis elit sint proident reprehenderit consectetur Lorem incididunt occaecat voluptate proident.'
};
