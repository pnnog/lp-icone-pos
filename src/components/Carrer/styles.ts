import styled, { DefaultTheme, css } from 'styled-components';
import media from 'styled-media-query';
import { CarrerProps } from '.';

type WrapperProps = Pick<CarrerProps, 'toRight'>;

const wrapperModifiers = {
  toRight: (theme: DefaultTheme) => css`
    grid-template-areas: 'description image';

    ${Description} {
      border-radius: ${theme.border.radius} 0 0 ${theme.border.radius};
    }
  `,
  toLeft: (theme: DefaultTheme) => css`
    grid-template-areas: 'image description';
    ${Description} {
      border-radius: 0 ${theme.border.radius} ${theme.border.radius} 0;
    }
  `
};

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, toRight }) => css`
    width: 100%;
    height: 100%;

    background: ${theme.colors.white};
    box-shadow: 0 11rem 3rem 0 rgba(143, 177, 200, 0.22);
    border-radius: ${theme.border.radius};

    display: grid;
    grid-template-columns: 1fr;

    ${media.greaterThan('medium')`
      ${!!toRight && wrapperModifiers.toRight(theme)};
      ${!toRight && wrapperModifiers.toLeft(theme)};
     
    `}
  `}
`;

export const ImageBox = styled.figure`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 25rem;
    align-self: end;
    padding: ${theme.spacings.xsmall} ${theme.spacings.xsmall} 0
      ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      grid-area: image;
      height: 30rem;
  `} img {
      max-width: 25rem;
      height: 100%;
    }

    ::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 0;
      height: 1rem;
      width: 100%;
      background: linear-gradient(
        90deg,
        rgba(12, 74, 110, 1) 18%,
        rgba(63, 147, 186, 1) 61%,
        rgba(233, 245, 247, 1) 100%
      );
    }

    ${media.greaterThan('medium')`
      ::after{
        height: 1.2rem;
        
      }
    `}
  `}
`;

export const Name = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.extraBold};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
      margin-bottom: ${theme.spacings.xxsmall};

    `}
  `}
`;
export const Role = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.regular};
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.medium};

    `}
  `}
`;

export const Description = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    padding: ${theme.spacings.small};
    border-radius: 0 0 ${theme.border.radius} ${theme.border.radius};

    h3,
    p,
    li {
      text-align: justify;
      color: ${theme.colors.white};
    }

    ul {
      display: flex;
      flex-direction: column;
      gap: ${theme.spacings.xsmall};
      list-style: none;
    }

    li {
      font-size: ${theme.font.sizes.small};
      font-weight: ${theme.font.regular};
      line-height: ${theme.font.sizes.xxlarge};
    }

    ${media.greaterThan('medium')`
      grid-area: description;
      li{
        font-size: ${theme.font.sizes.medium};
      }
    
    `}
  `}
`;
