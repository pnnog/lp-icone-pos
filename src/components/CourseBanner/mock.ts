export default {
  img: 'https://source.unsplash.com/user/willianjusten/1000x1000',
  title: 'Especialização em implantodontia',

  beginning: {
    title: 'Início',
    description: 'Março de 2023'
  },

  workload: {
    title: '1000 h/a',
    description: '26 módulos mensais'
  }
};
