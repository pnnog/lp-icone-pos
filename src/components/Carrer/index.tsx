import * as S from './styles';

export type CarrerProps = {
  img: string;
  name: string;
  role: string;
  description: string;
};

const Carrer = ({ img, name, role, description }: CarrerProps) => (
  <S.Wrapper>
    <S.ImageBox img={img} role="img" aria-label={name} />
    <S.Content>
      <S.Name> {name}</S.Name>
      <S.Role> {role}</S.Role>
      <S.Description>{description} </S.Description>
    </S.Content>
  </S.Wrapper>
);

export default Carrer;
