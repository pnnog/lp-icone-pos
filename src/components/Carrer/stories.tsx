import { Meta, Story } from '@storybook/react';
import Carrer, { CarrerProps } from '.';
import mock from './mock';

export default {
  title: 'Teachers/Carrer',
  component: Carrer,
  args: { ...mock },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta;

export const Basic: Story<CarrerProps> = (args) => (
  <div style={{ maxWidth: '103rem', margin: 'auto' }}>
    <Carrer {...args} />
  </div>
);
