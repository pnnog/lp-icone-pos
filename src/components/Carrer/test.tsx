import { screen } from '@testing-library/react';

import Carrer from '.';
import { renderWithTheme } from 'utils/tests/helper';

import mock from './mock';
import theme from 'styles/theme';

describe('<Carrer />', () => {
  it('should render Carrer', () => {
    renderWithTheme(<Carrer {...mock} />);

    expect(screen.getByRole('img', { name: mock.name })).toHaveAttribute('src', mock.img);

    expect(screen.getByRole('heading', { name: mock.name })).toHaveStyle({
      color: theme.colors.white
    });

    expect(screen.getByText(mock.role)).toHaveStyle({
      color: theme.colors.white
    });
  });

  it('should render pic on right', () => {
    const { container } = renderWithTheme(<Carrer {...mock} toRight />);

    expect(container.firstChild).toHaveStyleRule('grid-template-areas', "'description image'", {
      media: '(min-width: 768px)'
    });
  });

  it('should render pic on left', () => {
    const { container } = renderWithTheme(<Carrer {...mock} />);

    expect(container.firstChild).toHaveStyleRule('grid-template-areas', "'image description'", {
      media: '(min-width: 768px)'
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
