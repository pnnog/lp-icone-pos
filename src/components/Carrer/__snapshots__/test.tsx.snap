// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carrer /> should render pic on left 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background: none;
  border-radius: 0.4rem;
  display: grid;
}

.c1 {
  position: relative;
  width: 100%;
  height: 25rem;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  padding: 1.6rem 1.6rem 0 1.6rem;
}

.c1 img {
  max-width: 25rem;
  height: 100%;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  height: 1rem;
  width: 95%;
  background: linear-gradient( 90deg,rgba(12,74,110,1) 18%,rgba(63,147,186,1) 61%,rgba(233,245,247,1) 100% );
}

.c3 {
  font-size: 1.8rem;
  font-weight: 700;
}

.c4 {
  color: #475569;
  font-size: 1.2rem;
  font-weight: 400;
  margin-bottom: 2.4rem;
}

.c2 {
  box-shadow: 0 1rem 3rem 0 rgba(143,177,200,0.22);
  background-color: #0C4A6E;
  padding: 2.4rem;
  border-radius: 0 0 0.4rem 0.4rem;
}

.c2 h3,
.c2 p,
.c2 li {
  text-align: justify;
  color: #F8FAFC;
}

.c2 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
  list-style: none;
}

.c2 li {
  font-size: 1.4rem;
  font-weight: 400;
  line-height: 2.8rem;
}

@media (min-width:768px) {
  .c0 {
    grid-template-areas: 'image description';
  }
}

@media (min-width:768px) {
  .c1 {
    grid-area: image;
    height: 30rem;
  }
}

@media (min-width:768px) {
  .c1::after {
    height: 1.2rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.0rem;
    margin-bottom: 0.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.4rem;
    margin-bottom: 3.2rem;
  }
}

@media (min-width:768px) {
  .c2 {
    grid-area: description;
    border-radius: 0.4rem;
  }

  .c2 li {
    font-size: 1.6rem;
  }
}

<div
  class="c0"
>
  <figure
    class="c1"
  >
    <img
      alt="Doutor fulano"
      src="/img/float-1.png"
    />
  </figure>
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
       
      Doutor fulano
    </h3>
    <p
      class="c4"
    >
       
      Especialista e Mestre em Cirurgia Buco Maxilo Facial
    </p>
    <ul>
      <li>
        Professor do Mestrado de Odontologia nas áreas de Implantodontia e Estomatologia
      </li>
      <li>
        Especialista em Prótese dentária e Especialista em Implantodontia
      </li>
      <li>
        Professor da Escola Bahiana de Medicina e Saúde Pública, Curso de Odontologia
      </li>
      <li>
        Preceptor da Residência em Cirurgia Buco Maxilo Facial do Hospital Geral Roberto Santos/Bahiana
      </li>
    </ul>
  </div>
</div>
`;
